### ISSUE TITLE
<Suggestions when "unknown command" happens> only works at first level ?

### ISSUE THREAD

**Issue by dsgrangs-paul** [spf13/cobra#834]
Suggestions when "unknown command" happens

I have a three levels command, something like (presenting by level)  
 1/ mycmd 
 2/ mycmd list
    mycmd modify 

  3/ mycmd list volumes <flags>
     mycmd list disks <flags>
     mycmd list networks <flags>
     mycmd modify volumes <flags>
     mycmd modify disks <flags>
     mycmd modify networks <flags>

The suggestion works well for level 2 : if ever I do a typo on 'list' or 'modify' then I have the msg "Did you mean this?".
But the suggestions does not work at all if I do a typo on the third level : a typo on 'volumes', 'disks', 'networks'  does not generate the "Did you mean this?" message ?

I ve tried to play with such kind of code (see below), without success : 
	volumesCmd.DisableSuggestions = false
	volumesCmd.SuggestionsMinimumDistance = 2
	listCmd.AddCommand(volumesCmd)

Thank you for any help or clue !

**Comment by dpopp07** [comment]
I am also interested in this ^

From [this code](https://github.com/spf13/cobra/blob/a684a6d7f5e37385d954dd3b5a14fc6912c6ab9d/args.go#L15) it looks intentional but I'm wondering
1) why is that?
2) is there any way to make that configurable so that the lowest level commands (without subcommands) get suggestions for typos?

**Comment by johnSchnake** [comment]
I agree on the surface that looks like a bug but may have been intentional due to some legacy support. Added tags to issue but still in need of someone to pick it up.

**Comment by YuviGold** [comment]
Hi @johnSchnake I already implemented a solution on https://github.com/spf13/cobra/pull/1500 but the maintainers of the repository had mixed thoughts about it and didn't get to a consensus how to address this issue since it will break backward compatibility. 
@jpmcb let me know if you think it should be implemented or whether we can close this issue.
